lib/srv/discovery/discovery.go

```diff
@@ -1238,6 +1260,30 @@ func (s *Server) startDynamicWatcherUpdater() {
 	}
 }
 
+// newPollSubscription creates a new subscription for triggering a poll event.
+// The consumer must have an active reader on the returned channel, and start a new Poll when it returns a value.
+func (s *Server) newPollSubscription() (ch chan struct{}) {
```

Comment:
Can we add a bit context for this subscription 

```suggestion
func (s *Server) newDiscoveryConfigChangedSub() (ch chan struct{}) {
```